src/test/java/com/es/phoneshop/model/product/CustomProductDaoTest.java

```diff
@@ -0,0 +1,114 @@
+package com.es.phoneshop.model.product;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+public class CustomProductDaoTest {
+    private final CustomProductDao productDao = CustomProductDao.getInstance();
+
+    private List<Product> createValidProducts(int count) {
+        List<Product> productList = new ArrayList<>();
+        BigDecimal price = new BigDecimal(1);
+        int stock = 1;
+        for (int i = 1; i <= count; i++) {
+            Product product = new Product();
```

Comment:
senseless creation, this creates products with same attributes
please use @Mock for Products and mock their attributes to have different values